proto/cosmos/crypto/keyring/types.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+package cosmos.crypto.keyring;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+import "cosmos/crypto/hd/hd.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keyring";
+option (gogoproto.goproto_getters_all) = false;
+
+message Record {
+  string name = 1;
+  google.protobuf.Any pub_key = 2;
+
+  oneof item {
+    LocalInfo local = 3;
+    LedgerInfo ledger  = 4;
+    OfflineInfo offline  = 5;
+    MultiInfo multi  = 6;
+  }
+}
+
+// Item is a keyring item stored in a keyring backend.
+message LocalInfo {
+  google.protobuf.Any privKey = 1;
+  string pubKeyType = 2;
+}
+
+message LedgerInfo {
+  hd.BIP44Params path = 1;
+}
+
+message OfflineInfo {}
+
+message MultiInfo {}
```

Comment:
```suggestion
  oneof item {
    Local local = 3;
    Ledger ledger  = 4;
    Empty empty  = 5;
  }
  
  // Item is a keyring item stored in a keyring backend.
  message Local {
    google.protobuf.Any privKey = 1;
    string pubKeyType = 2;
  }

  message Ledger {
    hd.BIP44Params path = 1;
  }

  message Empty {}
}
```

Reply:
what do you think about moving the Record items into into the Record?

Reply:
you mean moving Record items into Record package?

Reply:
yes: Local, Ledger, Empty